src/Microsoft.DotNet.Build.Tasks/ParseTestCoverageInfo.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Xml;
+using System.Collections.Generic;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class ParseTestCoverageInfo : Task
+    {
+        // Path to the directory that contains *.coverage.xml coverage info.
+        [Required]
+        public string CoverageInfoDirectory { get; set; }
+
+        // One per xml parsed, key = test name, value list of modules visited
+        private Dictionary<string, Dictionary<string, ModuleInfo>> testCoverageInfo = null;
+
+        public override bool Execute()
+        {
+            if (!Directory.Exists(CoverageInfoDirectory))
+            {
+                Log.LogError("{0} is not a valid path", CoverageInfoDirectory);
+                return false;
+            }
+
+            if (!ProcessDirectory(CoverageInfoDirectory))
+            {
+                Log.LogError("There was no coverage file found.");
+                return false;
+            }
+
+            SaveToFile();
+
+            return true;
+        }
+
+        /// <summary>
+        /// Explore the input path recursively.
+        /// Parse opencoverage output.
+        /// </summary>
+        /// <param name="directory"></param>
+        /// <returns>True if at least one file that matched the pattern coverage_[name of test assembly].xml is found</returns>
+        private bool ProcessDirectory(string directory)
+        {
+            string[] results = Directory.GetFiles(directory, "*.coverage.xml", SearchOption.AllDirectories);
+
+            bool found = (results.Length > 0);
+
+            foreach (string file in results)
+            {
+                if (testCoverageInfo == null)
+                {
+                    testCoverageInfo = new Dictionary<string, Dictionary<string, ModuleInfo>>();
+                }
+
+                ParseCoverageFile(file);
+            }
+
+            return found;
+        }
+
+        /// <summary>
+        /// Parse all coverage .xml files
+        /// </summary>
+        private void ParseCoverageFile(string file)
```

Comment:
OpenCover doesn't include that. Even the tool that generates the .html reports implemented their custom parser. Although they use more Linq (https://github.com/danielpalme/ReportGenerator/blob/1deabff63aff52c419bbc811407cf7a0738c2262/ReportGenerator/Parser/OpenCoverParser.cs)

This report simplifies the OpenCover report so I think a custom one is appropiate.